### PR TITLE
Initial full buildroot-based distribution for tpu

### PR DESCRIPTION
**Pull request by jr1221**
This adds all the basics needed for tpu including:

- nrc7292 support + alfa leds and appropriate automatic networking for station mode on bootup (see bottom of readme for networking setup)
- raspberry pi 4 support on kernel 6.1
- nanomq with quic support
- gpsd + chrony for gps time support
- docker for future needs
- python for gpsd helper functions and future needs
- socketcand and can-utils (w/o setup or configuration)

A few custom scripts were needed, some of which replicated nrc7292 [start.py](https://github.com/newracom/nrc7292_sw_pkg/blob/master/package/evk/sw_pkg/nrc_pkg/script/start.py) functionality. This includes:
- a script which creates the parameters to pass into the driver (in utils folder).  It is mostly replicated from start.py
- a script which sets txpwr and guard interval on boot (initialize-nrc-extra in the overlayfs /usr/bin).
- init scripts for the leds, module loading, nanomq, and to load the extra nrc parameters (see above bullet).
- a script for the LEDs (ported to posix shell from the halow_setup folder of siren)

Current issues to be fixed in future PRs/upstream updates:  
- dhcpcd requires v10.0.5 due to [this issue](https://github.com/NetworkConfiguration/dhcpcd/issues/260) however buildroot is still on 10.0.5.  **Requires edit of buildroot source.**
- Buildroot rpi support was on kernel 5.10 and to use new SPI overlay kernel 5.16 was required.  Requires bumped buildroot HEAD to past version 2023.11 to gain the next kernel (v6.1).  When february release comes out change to stable branch.
- Nanomq cannot build as a shared library due to a multitude of errors (cause unknown, not a huge deal).
- 

More features to be added in future PRs:  
- correct SPI bus in overlays for both mcp can and nrc halow
- AP mode on a raspberry pi 3 (possibly??)
- nrc7394 support as part of the virtual packaging framework (depending on tpu hardware plans)
- socketcand autostart and correct configuration.

Most features require testing and they only work in theory rn.  Since systemd is so different from busybox networking may require more changes to get up and running, testing should occur immediately after break.

### CODE REVIEW

**Comment by jr1221** [comment]
Oh lol this commit is so big.  It is bc board/tpu/busybox.config and board/tpu/bcm2711_tpu_defconfig as they need one or two edits but it writes even default options to the file if you make any changes.  There is a README in that folder explaining generally what was changed.  It is too bad it isn't smart enough to strip out default options!

Also two problems I ran into:
- I change ccache location for myself and had to commit it as part of the tpu_defconfig
- I didn't specify an actual password in the defconfig for the pi

File: odysseus/odysseus_tree/utils/build_nrc_params.py [hunk 1]
```diff
@@ -0,0 +1 @@
+/home/jack/Projects/NER/buildroot/Siren/odysseus/odysseus_tree/overlays/rootfs_overlay_sta/usr/bin/build_nrc_params.py
```
Comment (nwdepatie) [inline odysseus/odysseus_tree/utils/build_nrc_params.py:1]: uh this is a local reference, tbh not sure how you can symlink in a git repo

**Review by nwdepatie — CHANGES_REQUESTED** [review]
I found a local reference for a symlink that we should prob fix, but other than that looks good, but I'm gonna try and build myself before approving. Very strong start tho

**Comment by jr1221** [comment]
Also renamed the board data file so that we don't confuse the upstream one and the ALFA specific one.

**Review by nwdepatie — APPROVED** [review]
Amazing work, this should make our implementation a LOT cleaner

**Pull request merged**